src/core/data/ions/ions.h

```diff
@@ -252,10 +252,23 @@ namespace core
                 }
             };
 
+
+            auto copy = [ix0, ix1](auto& pop) {
```

Comment:
I think I know what this does, but I don't know why.

Can there be a little comment stating why it's set to the border node values?

Reply:
sure, will add comment in code.

in the mean time, small explanation here.
We had NaN on all moments ghost nodes. Told you that I'd better have NaNs than "wrong" values there. Since none are used in the numerical algorithms, having NaNs there is/was a good idea because if one algo hit one of those nodes, we'd had seen NaNs in domain... and we'd know immediately something's wrong... 
Turns out the coarsening operator takes fine domain nodes and put values on coarse ones. To fill the coarse node lying aligned with the domain border fine node, one needs, for refinement factor 2, ONE ghost node. That ghost node being NaN it was setting NaNs in coarse domain nodes....

I chose to copy the first domain node value to the first ghost node so that no NaN is interpolated on coarse nodes.

all is explained here https://github.com/PHAREHUB/PHARE/issues/181